sktime/utils/_testing/deep_equals.py

```diff
@@ -114,6 +112,8 @@ def ret(is_equal, msg):
         return ret(*_tuple_equals(x, y, return_msg=True))
     elif isinstance(x, dict):
         return ret(*_dict_equals(x, y, return_msg=True))
+    elif type(x).__name__ == "ForecastingHorizon":
```

Comment:
Why not do `isinstance(x, ForecastingHorizon)` ?

Reply:
I wrote that in the PR message 😄 

> The extension avoids importing the `ForecastingHorizon`, ensuring that the `deep_equals` utility does not have an `sktime` dependency.
> (for reviewers: that's why `type(x).__name__` is used and not `isinstance`)

Reply:
> I wrote that in the PR message
🤦‍♂️ I must have run straight into the code, it was a bit late at night!

OK, I get it now. But why is it a problem to create an sktime dependency? 
The only use case I can think of is if someone copies the code.